src/core/Track.cpp

```diff
@@ -622,14 +624,10 @@ void Track::toggleSolo()
 				track->m_soloModel.setValue(false);
 			}
 		}
-		else if (!soloBefore)
+		// Revert all tracks to pre-solo values
+		else if (!otherTrackAlreadySolo)
 		{
-			// Unless we are on the sololegacybehavior mode, only restores the
-			// mute state if the track isn't an Automation Track
-			if (soloLegacyBehavior || track->type() != AutomationTrack)
-			{
-				track->setMuted(track->m_mutedBeforeSolo);
-			}
+			track->setMuted(track->m_mutedBeforeSolo);
```

Comment:
More thinking out loud:

If `soloLegacyBehavior` is false, then the automation track is not muted, and "restoring" it will just unmute it (no change).  So the if statement is not needed. This is correct.

(And for 1.2 projects `m_mutedBeforeSolo` defaults to false)